window-copy.c

```diff
@@ -3048,31 +3068,66 @@ window_copy_search_jump(struct window_mode_entry *wme, struct grid *gd,
 		return (window_copy_search_jump(wme, gd, sgd,
 		    direction ? 0 : gd->sx - 1,
 		    direction ? 0 : gd->hsize + gd->sy - 1, fy, cis, 0,
-		    direction, regex, foundlen));
+		    direction, regex));
 	}
 	return (0);
 }
 
+static void
+window_copy_move_before_search_mark(struct window_copy_mode_data *data,
+    u_int *fx, u_int *fy, int wrapflag)
+{
+	struct screen  *s = data->backing;
+	u_int		at, start;
+
+	if (window_copy_search_mark_at(data, *fx, *fy, &start) == 0 &&
+	    data->searchmark[start] != 0)
+		while (window_copy_search_mark_at(data, *fx, *fy, &at) == 0 &&
+		       at > 0) {
```

Comment:
I think we should drop the test for `at > 0`. It is true that there is no marking "before" `a==0` but there can be a position to move to before it if the mark starts at `a==0`. If you position a marked area to start at the top left of the visible screen and search up (easiest to check this in vi mode), you'll see that the search gets "stuck" there because of this check.

Reply:
Got it, updated.